### ISSUE TITLE
[Bug]: Blocked while sending a transaction, unable to respond

### ISSUE THREAD

**Issue by ginlink** [MetaMask/metamask-extension#13649]
### Describe the bug

Blocked while sending a transaction, unable to respond in versions greater than 10.9 ,
but in version 10.8.2, this problem does not occur
chainID: 9000
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/35068110/154212907-d9d66b2c-d0ac-42b0-b292-6e533715d55e.png">

### Steps to reproduce

1.switch to evmos network(9000)
2.click send
3.It was blocked after about 3 seconds

### Error messages or log output

```shell
gas price estimation failed due to network error
```

### Version

10.9.3,10.9.2

### Browser

Chrome, Firefox

### Operating system

MacOS

**Comment by darkwing** [comment]
Thank you for reporting this @ginlink .  I've tried getting test PHOTON from the faucet (https://faucet.evmos.org/) but it's currently failing, so I cannot immediately reproduce.  I will check again later today.

**Comment by ginlink** [comment]
@darkwing Thanks for your reply. There is something wrong with the faucet of Evmos. I have created a test account here.
Can I send the private key to you?

**Comment by ginlink** [comment]
@darkwing  Thanks for your reply. Just click on PHOTON coin and wait about 3 seconds, it will be blocked

**Comment by ginlink** [comment]
@darkwing Yes, there is something wrong with its faucet these days. If you need an account, I can create one for you and give you the private key.

**Comment by devli13** [comment]
Hey guys, I am from the Evmos team and just confirming this bug as we have been trying to work to resolve the past couple days. Would appreciate any thoughts as we have isolated the issue to Metamask specific as it does not reproduce on versions earlier than 10.10.0 and when trying curl commands we don't have any issues.

Something else that was found in the error logs is that this url which is called by metamask fails to return for Evmos:
Evmos: https://gas-api.metaswap.codefi.network/networks/9000/suggestedGasFees
Ethereum: https://gas-api.metaswap.codefi.network/networks/1/suggestedGasFees

 Happy to send funds to anyone in this thread if helpful but faucet should be back up

**Comment by 0x4a5e1e4baab** [comment]
Also looking at this issue and see request locking up at the same time as the UI, the request actually finally get initiated and completes in normal time.



![image](https://user-images.githubusercontent.com/4766349/156402116-00029195-c6c4-423d-b6b7-41f24593078d.png)

**Comment by loredanacirstea** [comment]
@darkwing , 
The issue (Metamask popup freezing when trying to do a transaction - like clicking on the `send` ETH button) is reproducible by returning an `eth_feeHistory` result with a `reward` array that contains `null` values. I tracked the problem down to this line `return { ...obj, [percentile]: fromHex(priorityFee) };`
https://github.com/MetaMask/controllers/blob/dd48bba89f3ae1a39fa5b170686c84bdf9ce0e3d/src/gas/fetchBlockFeeHistory.ts#L220
```
const priorityFeesByPercentile = percentiles.reduce(
    (obj, percentile, percentileIndex) => {
      const priorityFee = priorityFeesForEachPercentile[percentileIndex];
      return { ...obj, [percentile]: fromHex(priorityFee) };
    },
    {} as Record<Percentile, BN>,
  );
```

The `fromHex` function eventually uses this `stripHexPrefix` function, which throws an Error if the value is not a `string`:
https://github.com/ethereumjs/ethereumjs-monorepo/blob/58e5e0d059d55965b349c922af0ca6177bcb7f26/packages/util/src/internal.ts#L44-L49

I am not sure what happens next, I can only assume it enters some cycles when trying to stop/restart the polling in the GasFeeController.
A badly formatted value should not freeze the UI.

Proposed solution: check that `priorityFee` is of `string` type, otherwise use `0x0` (before `fromHex`). I can help with a PR fix if needed.

**Comment by danjm** [comment]
@loredanacirstea Thanks very much for researching this.

@mcmire What do you think about the solution proposed here by @loredanacirstea?

**Comment by mcmire** [comment]
Hi @loredanacirstea,

As you and others have pointed out, when we request gas estimates we first attempt to use an API that looks like `https://gas-api.metaswap.codefi.network/networks/{networkId}/suggestedGasFees`. If that fails we fall back to computing gas fee information based on `eth_feeHistory`.

In writing this fallback logic I used geth as a reference for how `eth_feeHistory` behaves. You can see [here](https://github.com/ethereum/go-ethereum/blob/afe9558bba80727f40ac214276b0c0bbb1b237a0/eth/gasprice/feehistory.go#L108) that it will account for a lack of information and use `0x0` in place of null. Now that I am revisiting this, I see that [this spec](https://github.com/ethereum/execution-apis/blob/2ccee32ad7a61848c9746b42e472c8e4df8810ff/src/eth/fee_market.json#L75-L89) also exists. The spec requires that all items in the `reward` array be numbers.

Therefore, I would rather chains be compatible with the spec rather than MetaMask account for deviations from the spec. The [fix to ethermint mentioned above](https://github.com/tharsis/ethermint/pull/970) seems like it would resolve this issue, is that correct? If so, I am not sure we would need to make a change to the `@metamask/controllers` package as you suggested.

That said, coming back to the original issue, I 100% agree that MetaMask shouldn't lock up if there are problems fetching gas estimates, and I think there are a few things that we can do (and have already done) to address this:

* Part of the reason this is probably happening is that if the fallback gas estimate logic (as I mentioned above) is triggered, it will retrieve a massive amount of block information via `eth_feeHistory`. It is likely that the UI froze because the network took a long time to respond with this data. We are about to issue another release will remove these extra network calls and this should make the UI more responsive.
* In addition, the request to https://gas-api.metaswap.codefi.network/networks/9000/suggestedGasFees should work. This API is run by another team and I will work with them to see what we can do to make this work for Evmos. That will prevent the fallback logic from even being triggered and should result in a snappy UI.
* We should also figure out a way to not block the UI if all of our attempts to fetch gas estimates in various ways fail. This is something we've discussed internally, and there are probably some designs we will need to implement to truly address this, but at the very least, we should probably just allow users to enter a custom max base fee and priority fee and continue.

**Comment by loredanacirstea** [comment]
@mcmire, about 
> In addition, the request to https://gas-api.metaswap.codefi.network/networks/9000/suggestedGasFees should work. This API is run by another team and I will work with them to see what we can do to make this work for Evmos.

What is needed from Evmos's side for this to work? If this is an open-source project, can you provide a link to it?
